crates/net/eth-wire/src/types/transactions.rs

```diff
@@ -47,23 +54,207 @@ where
 /// as the request's hashes. Hashes may be skipped, and the client should ensure that each body
 /// corresponds to a requested hash. Hashes may need to be re-requested if the bodies are not
 /// included in the response.
-#[derive_arbitrary(rlp, 10)]
+// #[derive_arbitrary(rlp, 10)]
 #[derive(Clone, Debug, PartialEq, Eq, RlpEncodableWrapper, RlpDecodableWrapper, Default)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 pub struct PooledTransactions(
     /// The transaction bodies, each of which should correspond to a requested hash.
-    pub Vec<TransactionSigned>,
+    pub Vec<PooledTransactionResponse>,
 );
 
 impl From<Vec<TransactionSigned>> for PooledTransactions {
     fn from(txs: Vec<TransactionSigned>) -> Self {
-        PooledTransactions(txs)
+        PooledTransactions(txs.into_iter().map(Into::into).collect())
+    }
+}
+
+/// A response to [`GetPooledTransactions`]. This can include either a blob transaction, or a
+/// non-4844 signed transaction.
+// TODO: redo arbitrary for this encoding - the previous encoding was incorrect. then this can be
+// replaced with `TransactionSigned` in `PooledTransactions`
+#[derive(Clone, Debug, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub enum PooledTransactionResponse {
```

Comment:
I think the naming can be confusing because this is not the response but only part of the response.

we should consider renaming this, but don't have good suggestions atm

Reply:
renamed to `PooledTransactionsElement`, maybe that's more clear